### PR TITLE
Fix compilation without MPI.

### LINKED ISSUE #1745: Build failure with WITH_MPI=false
http://cdash.kyomu.43-1.org/viewBuildError.php?buildid=13541

Merge commit 30728c46c9dd96c03b1b0f057575b3e5c7a9c8d8

### PR DESCRIPTION
**Pull request by bangerth**
This also fixes an issue where using std_cxx11::tuple did not work because
BOOST does not provide an operator< for it (C++11 appears to do so).

Fixes #1745 .

### CODE REVIEW

**Comment by bangerth** [comment]
@Rombur -- can you please take a look whether this makes sense?

File: source/grid/grid_tools.cc [hunk 1]
```diff
@@ -1650,7 +1650,7 @@ namespace GridTools
     // Create a local index for the locally "owned" vertices
     types::global_vertex_index next_index = 0;
     unsigned int max_cellid_size = 0;
-    std::set<std_cxx11::tuple<types::subdomain_id,types::global_vertex_index> > vertices_added;
+    std::set<std::pair<types::subdomain_id,types::global_vertex_index> > vertices_added;
```
Comment (Rombur) [inline source/grid/grid_tools.cc]: I don't think you can use ```std::get``` without C++11 so you also need to change those if you want to replace ```tuple``` by ```pair```.

**Comment by Rombur** [comment]
I think the easiest is just to guard this function so it always uses MPI. There is no need for it in serial and I am not sure you can even create a ```parallel::distributed::Triangulation``` in serial and this is the input parameter of the function. We could also force C++11 for this function.

File: source/grid/grid_tools.cc (hunk 1 above)
Comment (bangerth) [inline source/grid/grid_tools.cc]: It seemed to compile for me.

**Comment by bangerth** [comment]
Disabling the whole function would work for me. I'll adjust the patch.

Let's not make decisions about requiring C++11 on a per-function basis.

File: source/grid/grid_tools.cc (hunk 1 above)
Comment (Rombur) [inline source/grid/grid_tools.cc]: OK good.

**Comment by bangerth** [comment]
New patch is up here. I confirmed that it compiles without C++11 with and without MPI.

**Pull request merged**